### DISCUSSION TITLE
How to use spine 3.7 3.8 for axmol engine ?

### DISCUSSION THREAD

**Discussion by thienphuoc** [axmolengine/axmol#1200] [Q&A] [answered]
Currently, I see that the latest supported version is 4.1, but I have some resources that use versions 3.7 and 3.8, which are currently not working. I have tried to clone https://github.com/axmolengine/spine-axmol/tree/spine37 to build, but there are too many errors.

Please help me resolve this issues! Thanks!

![Screen Shot 2023-05-07 at 17 23 25](https://user-images.githubusercontent.com/17909394/236671951-38ff688b-71b8-42e5-a4a8-26b6349c3338.png)

**Comment by rh101** [comment]
>  I have tried to clone https://github.com/axmolengine/spine-axmol/tree/spine37 to build, but there are too many errors.

The engine-specific code for Spine was changed significantly to work with Cocos2d-x v4, and then updated again with bug fixes and improvements to work with Axmol.  ~~There is **no** way to use the older versions of the runtimes without back-porting the Axmol related code modifications to them.~~ (EDIT: halx99 has already updated them all!)

Is there any reason you can't update the 3.7 and 3.8 Spine models to use the latest 3.8 format?  Spine file exports from Spine v3.8.99 work in Axmol.

**Comment by halx99** [comment, marked as answer]
I update all branches of https://github.com/axmolengine/spine-axmol for compile with latest axmol, you can update and try again

**Reply by rh101** [reply]
I didn't realize that you had already updated multiple versions of the spine runtime to work with Axmol.  Thank you for doing that!

**Reply by thienphuoc** [reply]
thanks for your help
